opentracing-api/src/main/java/io/opentracing/SpanScheduler.java

```diff
@@ -0,0 +1,87 @@
+package io.opentracing;
+
+/**
+ * Scheduler allows an existing (possibly thread-local-aware) execution context provider to also manage an
+ * actively-scheduled OpenTracing Span.
+ * <p>
+ * In any execution context (or any thread, etc), there is at most one "active" Span primarily responsible for the
+ * work accomplished by the surrounding application code. That active Span may be accessed via the
+ * {@link Scheduler#active()} method. If the application needs to defer work that should be part of the same
+ * Span, the Scheduler provides a {@link Scheduler#captureActive()} method that returns a
+ * {@link Continuation}; this activation context may be used to re-activate and deactivate the
+ * captured Span in that other asynchronous executor and/or thread.
+ */
+public interface Scheduler {
+
+    /**
+     * A Continuation can be used *once* to activate a Span and other execution context, then deactivate once the
+     * active period has concluded. (In practice, this active period typically extends for the length of a deferred
+     * async closure invocation.)
+     * <p>
+     * Most users do not directly interact with Continuation, activate(), or deactivate(), but rather use
+     * Scheduler-aware Runnables/Callables/Executors. Those higher-level primitives need not be defined within the
+     * OpenTracing core API.
+     *
+     * @see Scheduler#capture(Span)
+     */
+    interface Continuation extends AutoCloseable {
+
+        /**
+         * Make the Span (and other execution context) encapsulated by this Continuation active and return it.
+         * <p>
+         * NOTE: It is an error to call activate() more than once on a single Continuation instance.
+         *
+         * @param finishOnDeactivate true if.f. the span should be finish()ed when the Continuation is deactivated
+         *
+         * @see Scheduler#capture(Span)
+         * @return the newly-activated Span
+         */
+        Span activate(boolean finishOnDeactivate);
```

Comment:
`finishOnDeactivate` really bums me out... without introducing a reference-counting requirement within the Span abstraction, I don't know how to avoid it.

PS: I tried the refcounting thing, and while it's "possible", it also adds a bunch of cognitive overhead and complexity for the API user. It still may be better than this. YMMV.